sdk/core/core/inc/az_json_read.h

```diff
@@ -66,15 +66,13 @@ typedef struct {
 
 az_json_state az_json_state_create(az_const_span const buffer);
 
-az_result az_json_read(az_json_state *const p_state, az_json_value *const out_value);
+az_result az_json_read(az_json_state * const p_state, az_json_value * const out_value);
```

Comment:
space before const ?  looks kind of weird to me..   is this for some standard formatting?

Reply:
This is really _subjective_, and arguments go both ways, so I am not insisting on this proposal. I understand that sticking it to name is more common in C world, which is already a strong argument against spacing it from both sides.
1. `int const * const` is easier to read than `int const *const x`, you clearly see both sides, and can get which CV-modifier applies where.
2. One could argue that you read `int *p` as "`int` is what you get when you write `*p`". However, this is not the case with `*const p`, and with C++ references: write `int &ref = n`, you write `&ref` and you won't get `int`. I know this is C SDK, but we might want to reuse the same style for consistency. In C++ world `int* x` is more common than in C, although `int *x` is also used. So, having it in between is a compromise, in my view.
3. Sticking `*` to name really matters in single-line multiple variable declarations (which we don't do), but even then there is no consistency:
`typedef int * int_ptr;
int x = 3;
int_ptr a = &x, b = &x; // now asterisk is the part of the type`
So, sometimes (only in single-line multiple variable declarations without typedefs?) it rather sticks to variable; and sometimes it sticks to type (typedefs); and in most cases you can't tell the difference - so again, I view it as compromise.

Feel free to tell me you still don't like it, even aesthetically - I will postpone this very setting from checking in as part of this PR. We can discuss it with the group, but I don't think I will have more supporting arguments than those above, plus I understand that we already have our code written using `int *p`'s, and I will likely not convince many enough people if overwhelming majority prefers only one space before asterisk, so I am not insisting with this setting.
However, if I convinced you, and there are no other objections to this PR, I'd check it in, but still would be open to discussion if someone dislikes it afterwards.

Reply:
As of this style being used anywhere else: it is used in boost (example: https://github.com/boostorg/smart_ptr/blob/274ec17836bfbfd85159ef3c358031b31fc3c6ea/include/boost/smart_ptr/shared_ptr.hpp#L217), however they are not very consistent. I've seen at least `int* p` and `int * p` in the same file there, and I won't be surprised if there are also some `int *p`'s in their code.

Reply:
I followed up with Victor offline, he is fine with me checking this in.